src/index.js

```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-exports.DAGNode = require('./dag-node.js')
+exports.DAGNode = require('./dag-node-factory.js')
```

Comment:
If we expose the DAGNode class, it'll be possible for users to use it, ie. use the constructor. The only way to force not to do that is to not expose it and that, like said, is the first step to an immutable data structure. This is a generally known and widely-used pattern: don't expose the data structure (for use) but rather provide an interface that *returns* the data structure (DAGNode in our case).

Is there a reason why we would *need* to expose it?